### PR TITLE
aws-sam-cli: 1.37.0 -> 1.48.0

### PR DESCRIPTION
**Pull request by tshaynik**
###### Description of changes

Bump version of aws-cli version to latest release, and the version of its dependency aws-sam-translator

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by kubukoz — APPROVED** [review]

File: pkgs/development/python-modules/aws-sam-translator/default.nix [hunk 1]
```diff
@@ -48,6 +48,7 @@ buildPythonPackage rec {
   disabledTests = [
     # AssertionError: Expected 7 errors, found 9:
     "test_errors_13_error_definitionuri"
+    "test_region_configuration"
```
Comment (turboMaCk) [inline pkgs/development/python-modules/aws-sam-translator/default.nix:51]: Can you add comment with reason why this test is being disabled?

Comment (tshaynik) [reply pkgs/development/python-modules/aws-sam-translator/default.nix:52]: Added a comment

Comment (turboMaCk) [reply pkgs/development/python-modules/aws-sam-translator/default.nix:52]: thanks. Please squash both commits into one and I think this can be merged.

**Comment by jiegec** [comment]
Superseded by https://github.com/NixOS/nixpkgs/pull/178302

File: pkgs/development/python-modules/aws-sam-translator/default.nix [hunk 2]
```diff
@@ -13,7 +13,7 @@
 
 buildPythonPackage rec {
   pname = "aws-sam-translator";
-  version = "1.42.0";
+  version = "1.45.0";
```
Comment (jonringer) [inline pkgs/development/python-modules/aws-sam-translator/default.nix:16]: The translator bump should be a separate commit

please look at https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes
